src/Nethermind/Nethermind.Core/Extensions/Bytes.Vector.cs

```diff
@@ -37,27 +37,71 @@ public static void Avx2Reverse256InPlace(Span<byte> bytes)
             }
         }
 
+        public static void Or(this Span<byte> thisSpan, Span<byte> valueSpan)
+        {
+            BinOp(thisSpan, valueSpan, false);
+        }
+
+        public static void Xor(this Span<byte> thisSpan, Span<byte> valueSpan)
+        {
+            BinOp(thisSpan, valueSpan, true);
+        }
+
+        public static uint CountBits(this Span<byte> thisSpan)
+        {
+            uint result = 0;
+            if (Popcnt.IsSupported)
+            {
+                Span<uint> uintSpam = MemoryMarshal.Cast<byte, uint>(thisSpan);
+                for (int i = 0; i < uintSpam.Length; i++)
+                {
+                    result += Popcnt.PopCount(uintSpam[i]);
+                }
+            }
+            else
+            {
+                for (int i = 0; i < thisSpan.Length; i++)
+                {
+                    int n = thisSpan[i];
+                    while (n > 0)
+                    {
+                        n &= n - 1;
+                        result++;
+                    }
+                }
+            }
+
+            return result;
+        }
+
         [MethodImpl(MethodImplOptions.AggressiveOptimization)]
-        public static void Or(this Span<byte> thisSpam, Span<byte> valueSpam)
+        private static void BinOp(this Span<byte> thisSpan, Span<byte> valueSpan, bool doXor)
         {
-            var length = thisSpam.Length;
-            if (length != valueSpam.Length)
+            var length = thisSpan.Length;
+            if (length != valueSpan.Length)
             {
                 throw new ArgumentException("Both byte spans has to be same length.");
             }
 
             int i = 0;
 
-            fixed (byte* thisPtr = thisSpam)
-            fixed (byte* valuePtr = valueSpam)
+            fixed (byte* thisPtr = thisSpan)
+            fixed (byte* valuePtr = valueSpan)
             {
                 if (Avx2.IsSupported)
                 {
                     for (; i < length - (Vector256<byte>.Count - 1); i += Vector256<byte>.Count)
                     {
                         Vector256<byte> b1 = Avx2.LoadVector256(thisPtr + i);
                         Vector256<byte> b2 = Avx2.LoadVector256(valuePtr + i);
-                        Avx2.Store(thisPtr + i, Avx2.Or(b1, b2));
+                        if (doXor)
+                        {
+                            Avx2.Store(thisPtr + i, Avx2.Xor(b1, b2));
+                        }
+                        else
+                        {
+                            Avx2.Store(thisPtr + i, Avx2.Or(b1, b2));
+                        }
```

Comment:
actually would prefer to have a seperate method for each. This now have to branch on each loop step, branchless code will be faster.